app/screens/settings/notification_email/notification_email.tsx

```diff
@@ -0,0 +1,210 @@
+import {useTheme} from '@context/theme';
+import useAndroidHardwareBackHandler from '@hooks/android_back_handler';
+import useNavButtonPressed from '@hooks/navigation_button_pressed';
+import {t} from '@i18n';
+import {popTopScreen, setButtons} from '@screens/navigation';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+import {typography} from '@utils/typography';
+import {getEmailInterval, getNotificationProps} from '@utils/user';
+
+import {getSaveButton} from '../config';
+import SettingBlock from '../setting_block';
+import SettingContainer from '../setting_container';
+import SettingOption from '../setting_option';
+import SettingSeparator from '../settings_separator';
+
+import type UserModel from '@typings/database/models/servers/user';
+
+const getStyleSheet = makeStyleSheetFromTheme((theme) => {
+    return {
+        disabled: {
+            color: changeOpacity(theme.centerChannelColor, 0.64),
+            ...typography('Body', 75, 'Regular'),
+        },
+    };
+});
+
+const emailHeaderText = {
+    id: t('notification_settings.email.send'),
+    defaultMessage: 'Send email notifications',
+};
+const emailFooterText = {
+    id: t('notification_settings.email.emailInfo'),
+    defaultMessage: 'Email notifications are sent for mentions and direct messages when you are offline or away for more than 5 minutes.',
+};
+
+const emailHeaderCRTText = {
+    id: t('notification_settings.email.crt.send'),
+    defaultMessage: 'Thread reply notifications',
+};
+const emailFooterCRTText = {
+    id: t('notification_settings.email.crt.emailInfo'),
+    defaultMessage: "When enabled, any reply to a thread you're following will send an email notification",
+};
+
+const SAVE_EMAIL_BUTTON_ID = 'settings_notification.email.save.button';
+
+type NotificationEmailProps = {
+    componentId: string;
+    currentUser: UserModel;
+    emailInterval: string;
+    enableEmailBatching: boolean;
+    isCRTEnabled: boolean;
+    sendEmailNotifications: boolean;
+}
+
+const NotificationEmail = ({componentId, currentUser, emailInterval, enableEmailBatching, isCRTEnabled, sendEmailNotifications}: NotificationEmailProps) => {
+    const notifyProps = useMemo(() => getNotificationProps(currentUser), [currentUser.notifyProps]);
+    const initialInterval = useMemo(() => getEmailInterval(
+        sendEmailNotifications && notifyProps?.email === 'true',
+        enableEmailBatching,
+        parseInt(emailInterval, 10),
+    ).toString(), []);
+    const initialEmailThreads = useMemo(() => Boolean(notifyProps?.email_threads === 'all'), []);
+
+    const [notifyInterval, setNotifyInterval] = useState<string>(initialInterval);
+    const [emailThreads, setEmailThreads] = useState(initialEmailThreads);
+
+    const intl = useIntl();
+    const serverUrl = useServerUrl();
+    const theme = useTheme();
+    const styles = getStyleSheet(theme);
+
+    const setEmailInterval = (interval: string) => {
+        setNotifyInterval(interval);
+    };
```

Comment:
Why is this needed? Can't you pass directly `setNotifyInterval` to wherever `setEmailInterval` is used?

Reply:
I guess the problem here is that action is `(string | boolean) => void`, while `setX` is `(string) => void` (and actually more stuff).

Reply:
fixed